### PR TITLE
Configurable connect and ping timeouts

### PR DESCRIPTION
**Pull request by Mystiflow**

### CODE REVIEW

File: api/src/main/java/net/md_5/bungee/api/ProxyConfig.java [hunk 1]
```diff
@@ -70,6 +70,20 @@
...
 
+    /**
+     * Time in milliseconds before timing out a clients server connect request
+     *
+     * @return connect timeout
+     */
+    int getConnectTimeout();
+
+    /**
+     * Time in milliseconds before timing out a clients ping
```
Comment (md-5) [inline api/src/main/java/net/md_5/bungee/api/ProxyConfig.java]: That's not what the option does -- probably #2753 relevant

**Comment by md-5** [comment]
This doesn't successfully compile, the unit tests fail.

File: api/src/main/java/net/md_5/bungee/api/ProxyConfig.java [hunk 2]
```diff
@@ -70,6 +70,21 @@
      */
     Collection<String> getDisabledCommands();
 
+    /**
+     * Time in milliseconds before timing out a clients request to connect to a server
```
Comment (md-5) [inline api/src/main/java/net/md_5/bungee/api/ProxyConfig.java]: Period at end of first sentence

Comment (md-5) [reply api/src/main/java/net/md_5/bungee/api/ProxyConfig.java]: 80 column docs

File: api/src/main/java/net/md_5/bungee/api/ProxyConfig.java [hunk 3]
```diff
@@ -70,6 +70,21 @@
...
+     *
+     * @return connect timeout
+     */
+    int getServerConnectTimeout();
+
+    /**
+     * Time in milliseconds before timing out a ping request from the proxy
+     * to a server when attempting to request server list info
```
Comment (md-5) [inline api/src/main/java/net/md_5/bungee/api/ProxyConfig.java]: As above

**Comment by Mystiflow** [comment]
I removed the unit test as I wasn't able to obtain a value for the default timeout without creating a dummy config of some sort

**Pull request closed without merging**